pkgs/development/libraries/mpich/default.nix

```diff
@@ -1,24 +1,50 @@
-{ lib, stdenv, fetchurl, perl, gfortran
-,  openssh, hwloc
+{ stdenv, lib, fetchurl, perl, gfortran
+,  openssh, hwloc, autoconf, automake, libtool
+# device options are ch3 or ch4
+, device ? "ch4"
+# backend option are libfabric or ucx
+, ch4backend ? "libfabric"
+, ucx, libfabric
+# Process manager to build
+, withPm ? "hydra:gforker"
 } :
 
 stdenv.mkDerivation  rec {
   pname = "mpich";
-  version = "3.3.2";
+  version = "3.4.1";
 
   src = fetchurl {
     url = "https://www.mpich.org/static/downloads/${version}/mpich-${version}.tar.gz";
-    sha256 = "1farz5zfx4cd0c3a0wb9pgfypzw0xxql1j1294z1sxslga1ziyjb";
+    sha256 = "09wpfw9lsrc84vcmfw94razd4xv4znx4mmg7rqmljvgg0jc96dl8";
   };
 
+  patches = [
+    # Reverts an upstream change that causes json-c test to fail
+    ./jsonc-test.patch
```

Comment:
Can you use fetchpatch and revert?

Reply:
Unfortunately not. The paths needed to be modified to match the mpich sub directory.

Reply:
fetchpatch can modify the path. Did you try that?

Reply:
Yes, I tried, but the a/ and b/ from the github patch gets inserted in the middle producing a garbled path name (the `stripLen` parameter does not solve the issue).